superduperdb/container/model.py

```diff
@@ -429,6 +434,20 @@ def __post_init__(self):
         else:
             self.to_call = getattr(self.object.artifact, self.predict_method)
 
+    @override
+    def on_load(self, db: DB) -> None:
+        if not self.to_method:
+            return
+
+        for device in self.preferred_devices:
+            try:
+                getattr(self.object.artifact, self.to_method)(device)
```

Comment:
Catching `Exception` will inevitable one day catch a programmer error!

`getattr(self.object.artifact, self.to_method)` should be extracted as a member in `__post_init__`, so that if `to_method` doesn't exist, you fail as early as possible.

Call it `self.artifact_to_method`.

Now, you're catching `self.artifact_to_method(device)`.  What exceptions are you actually expecting?  We should catch the most specific exception we can, and fail if there are any unexpected `Exception`s.

Reply:
I agree @duarteocarmo you can may be do 
except AttributeError

Reply:
Addressed the first. 

Regarding the second - [What changed? ](https://github.com/SuperDuperDB/superduperdb/pull/873#discussion_r1322012720)